Catena4430_cMeasurementLoop.cpp

```diff
@@ -209,22 +313,41 @@ cMeasurementLoop::fsmDispatch(
         if (fEntry)
             {
             // start SI1133 measurement (one-time)
-            this->m_si1133.start(true);
+            if (this->m_fSi1133)
+                this->m_si1133.start(true);
             this->updateSynchronousMeasurements();
             this->setTimer(1000);
             }
 
-        if (this->m_si1133.isOneTimeReady())
+        if (this->m_fSi1133)
```

Comment:
Make separate functions, not one gigantic function. Put all V1 code in a V1 function, all V2 code in a V2 function. We want to be able to throw out the V1 code in the future so it's necessary to segregate it as such. That means having a proper class deriviation:

```
cMeasurementLoop
    + cMeasurementLoop_4610v1
    + cMeasurementLoop_4610v2
```

The generic code defines virtual functions with no implementation `cMeasurementLoop::getTemperature` etc; then the concrete code does the work. Can't mix up v1 and v2!!!